### PR TITLE
Discord thread shutdown, last Autosave singlethreaded

### PR DESCRIPTION
**Pull request by rh-github-2015**
**Just an optional offer or food for thought** - feel free to scrap.

Discord looks much cleaner like this, can't hurt IMHO. 

The change of Autosave during dispose() solved truncated Autosave files for me - every debug session on a new map when it should have loaded an Autosave. When threaded it gets either aborted by the JVM before it finishes (despite not being a daemon), or its gameInfoClone breaks while it saves. Both sound unlikely, but the observed effect was there and is fixed by this.

I also left the thread enumerator in but now it only logs what definitely should be gone at that point, might help in the future. Though the hardcoded "DestroyJavaVM" thread name is likely to be JVM implementation dependent... The block should go into a #if DEBUG compiler conditional but I don't know how to do those in this ecosystem.

### CODE REVIEW

**Comment by yairm210** [comment]
Question: Is there a reason the cancelDiscord event is a separate function, outside of exitEvent? Are there situations where we would activate one but not the other?
If not, then I vote to include the closing of the discord thread into existing exitEvent of Deskop, that way we're not adding any extra functions etc.

File: desktop/src/com/unciv/app/desktop/DesktopLauncher.kt [hunk 1]
```diff
@@ -27,9 +30,9 @@ internal object DesktopLauncher {
         config.title = "Unciv"
         config.useHDPI = true
 
-        val versionFromJar = DesktopLauncher.javaClass.`package`.specificationVersion
+        val versionFromJar = DesktopLauncher.javaClass.`package`.specificationVersion ?: "Desktop"
```
Comment (yairm210) [inline desktop/src/com/unciv/app/desktop/DesktopLauncher.kt:33]: Much nicer, good one!

File: desktop/src/com/unciv/app/desktop/DesktopLauncher.kt [hunk 2]
```diff
@@ -71,27 +74,23 @@ internal object DesktopLauncher {
...
             Runtime.getRuntime().addShutdownHook(Thread { DiscordRPC.INSTANCE.Discord_Shutdown() })
 
-            thread {
-                while (true) {
-                    try {
-                        updateRpc(game)
-                    }catch (ex:Exception){}
-                    Thread.sleep(1000)
-                }
+            discordTimer = timer(name = "Discord", daemon = true, period = 1000) {
```
Comment (yairm210) [inline desktop/src/com/unciv/app/desktop/DesktopLauncher.kt:83]: Also much better, I feel dumb for not doing that in the first place now =/

**Comment by yairm210** [comment]
Apart from that minor comment, this looks good to go =)

**Comment by rh-github-2015** [comment]
Because exitEvent never gets called at all on desktop? Maybe on some keyboards, but not on mine. Don't confuse BACK (4) with BACKSPACE (67) - the one above Enter. And CameraStageBaseScreen.onBackButtonClicked is the only place that uses exitEvent.

**Pull request merged**